dali/c_api/c_api.h

```diff
@@ -69,6 +75,44 @@ extern "C" {
       int cpu_prefetch_queue_depth,
       int gpu_prefetch_queue_depth);
 
+  /**
+   * @brief Feed the data to ExternalSource as contiguous memory.
```

Comment:
I'd propose to put here a remark about performence differences between these two functions, or no perf diffs, if there are none. E.g. it might be possible that user can choose, if he want to put the samples into contiguous mem himself, if he can do it efficiently

Reply:
I think most we should do is point which C++ Pipeline API call are we using underneath. This API is just thin wrapper over that.

Reply:
Whatever we do here, I'm just pointing out, that from user's perspective this is quite important information, what's the actual difference (or if there's no difference) apart from convinency

Reply:
TBH, the answer to that is a little bit complicated and it would also change additionally if we allow the GPU inputs. The Pipeline doc doesn't provide such information.

Reply:
The choice is rather not about the performance but how your data looks like - if it is contiguous or not. I don't think anyone would convert continuous<-> noncontinuous just to use one of these functions.

Reply:
I think so far we're the only user of this API.
We know it's broken.
The function returns void and should return an error code.
There are more issues - this adds up to one thing: this signature is not stable anyway, so we can add the paramters in the next iteration, when the signature changes anyway.